capture_service/dive_client_cli.cc

```diff
@@ -22,421 +22,389 @@ limitations under the License.
 #include <system_error>
 #include <thread>
 
+#include "absl/base/no_destructor.h"
 #include "absl/flags/flag.h"
 #include "absl/flags/internal/flag.h"
 #include "absl/flags/parse.h"
 #include "absl/flags/usage.h"
 #include "absl/flags/usage_config.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 #include "absl/strings/str_split.h"
+
 #include "android_application.h"
 #include "command_utils.h"
 #include "constants.h"
 #include "device_mgr.h"
 #include "network/tcp_client.h"
-#include "absl/strings/str_cat.h"
 #include "utils/version_info.h"
 
-using namespace std::chrono_literals;
+enum class AppType
+{
+    kVulkan_OpenXR,         // OpenXR Vulkan app
+    kVulkan_Non_OpenXR,     // Vulkan app
+    kVulkanCLI_Non_OpenXR,  // Vulkan command line app
+    kGLES_OpenXR,           // OpenXR GLES app
+};
+
+struct GlobalOptions
+{
+    std::string serial;
+    std::string package;
+    std::string vulkan_command;
+    std::string vulkan_command_args;
+    AppType     app_type;
+    std::string device_architecture;
+    std::string download_dir;
+    std::string gfxr_capture_file_dir;
+    int         trigger_capture_after;
+
+    Dive::GfxrReplaySettings replay_settings;
+};
+
+struct CommandContext
+{
+    Dive::DeviceManager& mgr;
+    const GlobalOptions& options;
+};
 
 enum class Command
 {
     kNone,
-    kGfxrCapture,
-    kGfxrReplay,
     kListDevice,
     kListPackage,
     kRunPackage,
-    kRunAndCapture,
+    kPm4Capture,
+    kGfxrCapture,
+    kGfxrReplay,
     kCleanup,
 };
 
-bool AbslParseFlag(absl::string_view text, Command* command, std::string* error)
+struct CommandMetadata
 {
-    if (text == "list_device")
-    {
-        *command = Command::kListDevice;
-        return true;
-    }
-    if (text == "list_package")
-    {
-        *command = Command::kListPackage;
-        return true;
-    }
-    if (text == "run")
-    {
-        *command = Command::kRunPackage;
-        return true;
-    }
-    if (text == "capture")
-    {
-        *command = Command::kRunAndCapture;
-        return true;
-    }
-    if (text == "cleanup")
-    {
-        *command = Command::kCleanup;
-        return true;
-    }
-    if (text == "gfxr_capture")
-    {
-        *command = Command::kGfxrCapture;
-        return true;
-    }
-    if (text == "gfxr_replay")
-    {
-        *command = Command::kGfxrReplay;
-        return true;
-    }
-    if (text.empty())
+    std::string                                        name;
+    std::string                                        description;
+    std::function<absl::Status(const GlobalOptions&)>  validator;
+    std::function<absl::Status(const CommandContext&)> executor;
+};
+
+namespace
+{
+
+// Helper to validate options common to run/capture commands.
+absl::Status ValidateRunOptions(const GlobalOptions& options)
+{
+    if (options.package.empty() && options.vulkan_command.empty())
```

Comment:
would it be better if we add checks like:
- if vulkan_command is used, the app_type must be kVulkanCLI_Non_OpenXR. 
- If the command is present but the type is kVulkan_OpenXR, it is likely wrong.